### PR TITLE
[fix] update CMakeLists to copy QSQL driver on Apple

### PR DESCRIPTION
**Pull request by poshul**
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by poshul** [comment]
This gets a lot further with the changes to the mac build scripts, but still fails during the packing step with the error:
`CPack Error: Error executing: osascript "/Users/builder/jenkins/ws/openms/usr/pkg/87227eb0/bld/macOS_bundle_setup/setup_applescript.scpt" "OpenMS-3.0.0-pre-HEAD-2022-10-27-macOS"
CPack Error: Error executing custom script on disk image.
/Users/builder/jenkins/ws/openms/usr/pkg/87227eb0/bld/macOS_bundle_setup/setup_applescript.scpt:643:692: execution error: Finder got an error: AppleEvent timed out. (-1712)
`
@jpfeuffer I'm really confused by this since the script is only 89 lines, and this says it fails on 643.

**Comment by poshul** [comment]
This should be ready to go, but I can't test it with CI since the package build in usr is broken.

File: cmake/package_dragndrop_dmg.cmake [hunk 1]
```diff
@@ -50,7 +50,7 @@ set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${OPENMS_PACKAGE_VERSION_FULL
 ## Fix OpenMS dependencies for all executables in the install directory under bin.
 ## That affects everything but the bundles (which have their own structure and are fixed up in add_mac_bundle.cmake)
 ########################################################### Fix Dependencies
-install(CODE "execute_process(COMMAND ${PROJECT_SOURCE_DIR}/cmake/MacOSX/fix_dependencies.rb -b \${CMAKE_INSTALL_PREFIX}/${INSTALL_BIN_DIR}/ -l \${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}/ )"
+install(CODE "execute_process(COMMAND ${PROJECT_SOURCE_DIR}/cmake/MacOSX/fix_dependencies.rb -b \${CMAKE_INSTALL_PREFIX}/${INSTALL_BIN_DIR}/ -l \${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}/ -p \${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}/plugins/plugins/ )"
```
Comment (jpfeuffer) [inline cmake/package_dragndrop_dmg.cmake:53]: The -p option is not really well tested. Prepare for failures with weird rpaths or relative install_names.

**Comment by jpfeuffer** [comment]
Why is it broken?

**Comment by poshul** [comment]
> Why is it broken?

It gives an `AppleEvent timed out` error. I posted a bit about it a couple of comments up.

File: cmake/package_dragndrop_dmg.cmake (hunk 1 above)
Comment (jpfeuffer) [reply cmake/package_dragndrop_dmg.cmake:53]: I fixed it mostly in my Knime PR

**Comment by jpfeuffer** [comment]
Which checks did you perform?

**Comment by poshul** [comment]
The DMG builds and installs on Abibuilder1. Trying to actually run a workflow with an OMS file gives me errors because there are two QSQLITE instances available. Can you try loading the DMG on your mac @jpfeuffer https://abibuilder.cs.uni-tuebingen.de/jenkins/job/openms/job/usr/job/pkg/compiler=appleclang-11.0.0,os_label=catalina/lastSuccessfulBuild/artifact/bld/OpenMS-3.0.0-pre-HEAD-2022-11-03-macOS.dmg

**Comment by jpfeuffer** [comment]
Hmm even if it works for me, it should especially work on the same computer. I think this needs to be fixed.

**Comment by jpfeuffer** [comment]
What is the exact error?

**Comment by poshul** [comment]
This is the error.
image

**Comment by jpfeuffer** [comment]
It finds two QtCore and NO QSqlite. So we have two errors.

**Comment by poshul** [comment]
> It finds two QtCore and NO QSqlite. So we have two errors.
This is outside of my area of expertise, but it looks to me like it tries to load QtCore, can't because it's loading the wrong version. Then tries to load QSQLITE driver, which it sees, `ASqlDatabase: available drivers: QSQLITE`, but can't load because it failed to load QtCore supra.

**Comment by cbielow** [comment]
going on a disruptive tangent here -- sorry in advance -- but: would the time spent on trying to fix Qt deployment have been sufficient to just switch to another SQL lib?

**Comment by poshul** [comment]
> 

In retrospect, yes, the time spent on the Windows side was worthwhile at least for me to gain a good concept of how that build system works. I would argue here that the bigger issue is that Apple actively makes their systems difficult to develop for (restricting how they can be packaged, restricting signing, making certification pay-to-play, etc.).

I spent some time working with @jpfeuffer on this earlier, and we were able to get it working for the command line apps, but not the app bundles. I think at this point that is going to have to suffice, and we can discuss continuing support for Mac in general later.

**Comment by jpfeuffer** [comment]
https://bugreports.qt.io/browse/QTBUG-108211

**Pull request closed without merging**